Compile error (warning) on Linux

The compiler (GCC 11.2.0) complains that `hints` might not have been:

https://github.com/openthread/openthread/blame/main/src/posix/platform/infra_if.cpp#L647

```
third_party/openthread/repo/src/posix/platform/infra_if.cpp
third_party/openthread/repo/src/posix/platform/infra_if.cpp: In member function ‘otError ot::Posix::InfraNetif::DiscoverNat64Prefix(uint32_t)’:
third_party/openthread/repo/src/posix/platform/infra_if.cpp:647:21: error: ‘hints’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  647 |         freeaddrinfo(hints);
      |         ~~~~~~~~~~~~^~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
```

This will be the case if this check fails: https://github.com/openthread/openthread/blame/main/src/posix/platform/infra_if.cpp#L616.

--- a/src/posix/platform/infra_if.cpp
+++ b/src/posix/platform/infra_if.cpp
@@ -615,7 +615,7 @@ exit:
 otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
 {
     otError          error = OT_ERROR_NONE;
-    struct addrinfo *hints;
+    struct addrinfo *hints = nullptr;
     struct gaicb *   reqs[1];
     struct sigevent  sig;
     int              status;
@@ -649,7 +649,10 @@ otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
 exit:
     if (error != OT_ERROR_NONE)
     {
-        freeaddrinfo(hints);
+        if (hints)
+        {
+            freeaddrinfo(hints);
+        }
         free(reqs[0]);
     }
     return error;

